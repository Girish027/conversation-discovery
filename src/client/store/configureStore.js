import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import { routerMiddleware } from 'connected-react-router/immutable';
import {
  fromJS,
} from 'immutable';
import createRootReducer from './reducers';

const loggerMiddleware = createLogger({
  predicate: () => process.env.NODE_ENV === 'development',
  stateTransformer: (state) => JSON.parse(JSON.stringify(state)),
  actionTransformer: (action) => JSON.parse(JSON.stringify(action)),
});


const configureStore = (initialState = {}, history) => {
  const middlewares = [
    routerMiddleware(history), // for dispatching history actions
    thunk,
    loggerMiddleware
  ];

  const store = createStore(
    createRootReducer(history),
    fromJS(initialState),
    composeWithDevTools(applyMiddleware(...middlewares))
  );

  // Hot reloading
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createRootReducer(history));
    });
  }

  return store;
};

export default configureStore;
