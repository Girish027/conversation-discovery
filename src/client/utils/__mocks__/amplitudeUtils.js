const amplitudeUtils = jest.genMockFromModule('../amplitudeUtils');

amplitudeUtils.initializeAnalytics = jest.fn((analyticsKey, userName = '') => 'called initializeAnalytics');

amplitudeUtils.logAmplitudeEvent = jest.fn((eventName, eventSpecificData = {}) => 'called logAmplitudeEvent');

module.exports = amplitudeUtils;
