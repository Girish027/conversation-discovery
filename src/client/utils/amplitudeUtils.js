// import amplitude from 'amplitude-js/amplitude';

export const initializeAnalytics = (analyticsKey, userName = '') => {
  // const analytics = amplitude.getInstance();
  // analytics.init(analyticsKey, userName);
};

export const logAmplitudeEvent = (eventName, eventSpecificData = {}) => {
  // amplitude.getInstance().logEvent(eventName, {
  //   ...eventSpecificData,
  // });
};
