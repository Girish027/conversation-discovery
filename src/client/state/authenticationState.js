import {
  fromJS, Map
} from 'immutable';
import { createSelector } from 'reselect';
import axios from 'axios'; // this module is Promise based HTTP client for the browser and node.js
import appLogger from 'utils/appLogger';
import { getUserConfig } from 'state/appState';
import constants from '../constants';
import { initializeAnalytics, logAmplitudeEvent } from '../utils/amplitudeUtils';

export const types = {
  AUTHENTICATION_COMPLETE: 'auth/AUTHENTICATION_COMPLETE',
  LOGGED_OUT: 'auth/LOGGED_OUT'
};

export const stateKey = {
  name: 'name',
  oktaUrl: 'oktaUrl',
  email: 'email'
};

const devlog = appLogger({ tag: 'authenticationState' });

export const initialAuthState = fromJS({
  name: 'User Name',
  oktaUrl: ''
});

export const actionCompletedAuth = (userInfo) => ({
  type: types.AUTHENTICATION_COMPLETE,
  userInfo
});

export const actionLoggedoutAuth = () => ({
  type: types.LOGGED_OUT
});

export const getAuthUserinfo = () => (dispatch) => {
  axios.get(constants.serverApiUrls.authentications)
    .then((res) => {
      const authInfoArray = res.data;
      if (Array.isArray(authInfoArray) && authInfoArray.length) {
        const authInfo = authInfoArray[0];
        const userInfo = {
          name: authInfo.name,
          oktaUrl: authInfo.oktaUrl,
          email: authInfo.email,
        };
        initializeAnalytics(authInfo.analyticsKey, authInfo.name);
        const IDTUserEvent = { username: authInfo.name, toolId: constants.toolName, env: constants.environment };
        logAmplitudeEvent('IDTUser', IDTUserEvent);
        dispatch(actionCompletedAuth(userInfo));
        dispatch(getUserConfig());
      } else {
        dispatch(actionLoggedoutAuth());
      }
    })
    .catch((fetchErr) => {
      devlog.error(fetchErr);
    });
};

// REDUCER
export const authReducer = (state = initialAuthState, action = {}) => {
  switch (action.type) {
    case types.AUTHENTICATION_COMPLETE:
      return state.merge(Map(action.userInfo));
    case types.LOGGED_OUT:
      return initialAuthState;
    default:
      return state;
  }
};

// SELECTORS
export const getAuthState = (state) => state.get('auth');

export const getAuthNameSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.name),
);

export const getOktaUrlSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.oktaUrl),
);

export const getEmailSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.email),
);
