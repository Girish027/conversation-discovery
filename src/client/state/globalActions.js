import { downloadFile } from 'utils/FileUtils';
import { getCAASelector } from 'state/appState';
import { getActiveProjectId } from 'state/projectsState';
import Constants from '../constants';
import { logAmplitudeEvent } from '../utils/amplitudeUtils';
import {
  DOWNLOAD_FILE
} from './types';

export const downloadResults = (constructURL, runId, clusterId = '') => (dispatch, getState) => {
  const state = getState();
  const projectId = getActiveProjectId(state);
  const caa = getCAASelector(state);
  const locationUrl = constructURL({
    ...caa, projectId, runId, clusterId
  });
  dispatch({ type: DOWNLOAD_FILE, locationUrl });
  const downloadRun = { toolId: Constants.toolName, runId, env: Constants.environment };
  logAmplitudeEvent('IDT_DownloadRunResults', downloadRun);
  downloadFile(locationUrl);
};
