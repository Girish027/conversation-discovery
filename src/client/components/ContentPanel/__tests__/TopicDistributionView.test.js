import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ConnectedTopicDistributionView, { TopicDistributionView } from 'components/ContentPanel/TopicDistributionView';
import { fromJS } from 'immutable';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Constants from 'components/../constants';
import { runActions as runState } from 'state/runsState';
import { clusterActions as clusterState } from 'state/clusterState';
import * as globalActions from 'state/globalActions';
import * as amplitudeUtils from '../../../utils/amplitudeUtils';

jest.mock('utils/amplitudeUtils');
jest.mock('state/globalActions');

describe('<TopicDistributionView />', function () {
  let wrapper;
  let store;

  const caa = {
    clientId: '247inc',
    appId: 'referencebot',
    accountId: 'referencebot',
  };

  const runsState = {
    trackRuns: false,
    selectedRunId: 'run-59da2519-ed17-46ad-8294-1367e287589d',
    runs: [{
      runId: 'run-59da2519-ed17-46ad-8294-1367e287589d',
      projectId: 'pro-3e4c72f8-72cc-4e2d-77bb-11a5a266514c',
      runName: 'test-1564166348646',
      runDescription: 'trial',
      numOfTurns: 4,
      numOfClusters: 400,
      stopWords: '["hello","hilton"]',
      modified: 1564166848680,
      modifiedBy: 'abc@test.com',
      created: 1564166348680,
      createdBy: 'abc@test.com',
      runStatus: 'QUEUED',
      runStatusDescription: 'The run is queued',
      starred: 0
    }]
  };

  const projectsState = {
    activeProjectId: 'pro-3e4c72f8-72cc-4e2d-77bb-11a5a266514c',
    projects: [{
      caaId: '247ai-referencebot-referencebot',
      created: 1563227027435,
      createdBy: 'user@247.ai',
      datasetName: 'test.csv',
      modified: 1563227027435,
      modifiedBy: 'user@247.ai',
      projectDescription: 'project A desc',
      projectId: 'pro-3e4c72f8-72cc-4e2d-77bb-11a5a266514c',
      projectName: 'project A',
      projectStatus: 'READY',
      projectStatusDescription: null,
    }]
  };

  const clustersState = {
    selectedClusterId: '',
    clusters: [{
      clusterId: '1',
      clusterName: 'cancel_order_customer_service', // this is the one getting edited
      originalName: 'cancel_order_customer_service',
      clusterDescription: 'relates to cancel order',
      rollupCluster: 'Cancel_Order',
      suggestedNames: ['cancel_order_customer_service', 'abcde', 'defg'],
      count: 230,
      modifiedBy: 'user1@247.ai',
      modified: 1554930400000,
      finalized: false,
      finalizedOn: undefined,
      finalizedBy: '',
    }, {
      clusterId: '2',
      clusterName: 'cancel_order_service',
      originalName: 'Cancel_Order',
      clusterDescription: 'order canceled',
      rollupCluster: 'Cancel_Order',
      suggestedNames: ['cancel_order_service', 'abcde', 'defg'],
      count: 400,
      modifiedBy: 'user1@247.ai',
      modified: 1554930400000,
      finalized: false,
      finalizedOn: undefined,
      finalizedBy: '',
    }, {
      clusterId: '3',
      clusterName: 'store_close_holiday',
      originalName: 'store_close',
      clusterDescription: 'store closed due to reasons',
      rollupCluster: 'Store_Close',
      suggestedNames: ['merchant_close', 'abcde', 'defg'],
      count: 218,
      modifiedBy: 'user1@247.ai',
      modified: 1554930400000,
      finalized: false,
      finalizedOn: undefined,
      finalizedBy: '',
    }, {
      clusterId: '4',
      clusterName: 'size_and_position',
      originalName: 'size_and_position',
      clusterDescription: 'relates to size and position',
      rollupCluster: 'Store_Location',
      suggestedNames: ['position_size', 'store_location', 'defg'],
      count: 142,
      modifiedBy: 'user1@247.ai',
      modified: 1554930400000,
      finalized: true,
      finalizedOn: undefined,
      finalizedBy: 'user1@247.ai',
    }]
  };

  const props = {
    clusters: fromJS(clustersState.clusters),
    selectedRunId: runsState.selectedRunId,
  };

  beforeAll(() => {
    // Mock store
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    store = mockStore(fromJS({
      app: {
        ...caa,
      },
      projects: projectsState,
      runs: runsState,
      clusters: clustersState,
    }));
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getConnectedWrapper = () => mount(
    <Provider store={store}>
      <ConnectedTopicDistributionView />
    </Provider>);

  const getShallowWrapper = (propsObj) => shallow(<TopicDistributionView
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist - connected component TopicDistributionView', () => {
      wrapper = getConnectedWrapper();
      expect(wrapper.exists()).toBe(true);
    });

    test('should exist - basic TopicDistributionView component', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly when there are no clusters yet', () => {
      wrapper = getShallowWrapper({});
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders Rollup Distribution when there are clusters yet', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders Granular distribution for all clusters', () => {
      wrapper = getShallowWrapper(props);
      wrapper.setState({
        showRollupGraph: false
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders Granular distribution for the selectedRollup cluster', () => {
      wrapper = getShallowWrapper(props);
      wrapper.setState({
        showRollupGraph: false,
        selectedRollup: 'Cancel_Order'
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when finalized filter is applied', () => {
      const filters = {
        finalized: true
      };
      const newProps = { ...props, filters: fromJS(filters) };
      wrapper = getShallowWrapper(newProps);
      wrapper.setState({
        showRollupGraph: false,
        selectedRollup: 'Cancel_Order'
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when open filter is applied', () => {
      const filters = {
        open: true
      };
      const newProps = { ...props, filters: fromJS(filters) };
      wrapper = getShallowWrapper(newProps);
      wrapper.setState({
        showRollupGraph: false,
        selectedRollup: 'Cancel_Order'
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    let chartWrapper = {};
    let getSelection;
    beforeAll(() => {
      props.dispatch = jest.fn();
      clusterState.selectCluster = jest.fn(() => 'selectCluster');
      globalActions.downloadResults = jest.fn();
      runState.editRun = jest.fn(() => 'editRun');
    });

    beforeEach(() => {
      wrapper = getShallowWrapper(props);
    });

    afterEach(() => {
      jest.clearAllMocks();
      store.clearActions();
    });

    test('onSelectCluster', () => {
      wrapper.instance().onSelectCluster('2');
      expect(clusterState.selectCluster).toHaveBeenCalledWith('2');
      expect(props.dispatch).toHaveBeenCalledWith('selectCluster');
    });

    test('onClickGranularBar: for all granular cluster distribution', () => {
      getSelection = jest.fn(() => [{ row: 2, column: 1 }]);
      chartWrapper = {
        getChart: () => ({
          getSelection
        })
      };
      wrapper.instance().onClickGranularBar({ chartWrapper });
      expect(clusterState.selectCluster).toHaveBeenCalledWith('3');
      expect(props.dispatch).toHaveBeenCalledWith('selectCluster');
    });

    test('onClickGranularBar: for rollup level granular cluster distribution', () => {
      getSelection = jest.fn(() => [{ row: 1, column: 1 }]);
      chartWrapper = {
        getChart: () => ({
          getSelection
        })
      };
      wrapper.setState({
        selectedRollup: 'Cancel_Order'
      });
      wrapper.instance().onClickGranularBar({ chartWrapper });
      expect(clusterState.selectCluster).toHaveBeenCalledWith('1');
      expect(props.dispatch).toHaveBeenCalledWith('selectCluster');
    });

    test('toggleGraphType', () => {
      expect(wrapper.state().showPieChart).toEqual(true);
      wrapper.instance().toggleGraphType();
      expect(amplitudeUtils.logAmplitudeEvent).toHaveBeenCalledTimes(1);
      expect(wrapper.state().showPieChart).toEqual(false);
      // expect(wrapper.state().selectedRollup).toEqual('');
    });

    test('markAsComplete', () => {
      wrapper.instance().markAsComplete();
      expect(runState.editRun).toHaveBeenCalledWith(props.selectedRunId, {
        runStatus: Constants.status.COMPLETE,
        runStatusDescription: expect.any(String)
      }, false);
      expect(props.dispatch).toHaveBeenCalledWith('editRun');
    });

    test('downloadResults', () => {
      globalActions.downloadResults.mockReturnValue('downloadResults');
      wrapper.instance().downloadRunResults();
      expect(props.dispatch).toHaveBeenCalledWith('downloadResults');
      expect(globalActions.downloadResults).toHaveBeenCalled();
    });
  });
});
