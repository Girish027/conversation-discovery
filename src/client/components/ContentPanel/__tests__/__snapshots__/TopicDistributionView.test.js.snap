// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicDistributionView /> Snapshots renders Granular distribution for all clusters 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "0px 3px",
      }
    }
  >
    <div
      style={Object {}}
    >
      <Chart
        chartActions={null}
        chartEvents={
          Array [
            Object {
              "callback": [Function],
              "eventName": "ready",
            },
            Object {
              "callback": [Function],
              "eventName": "select",
            },
          ]
        }
        chartType="PieChart"
        chartWrapperParams={Object {}}
        className=""
        columns={null}
        controls={null}
        data={
          Array [
            Array [
              "Topic",
              "Volume",
              Object {
                "role": "wordCloudTerms",
              },
            ],
            Array [
              "cancel_order_service",
              400,
              undefined,
            ],
            Array [
              "cancel_order_customer_service",
              230,
              undefined,
            ],
            Array [
              "store_close_holiday",
              218,
              undefined,
            ],
            Array [
              "size_and_position",
              142,
              undefined,
            ],
          ]
        }
        diffdata={null}
        formatters={null}
        getChartEditor={null}
        getChartWrapper={[Function]}
        graphID={null}
        graph_id={null}
        height="calc(100vh - 40px - 90px - 67px - 10px)"
        legendToggle={false}
        legend_toggle={false}
        loader={
          <Placeholder
            styleOverride={Object {}}
          >
            <div
              className="message-default"
            >
              <LoadingIndicator
                color="#A7A9AF"
                isHerbie={true}
                message="Preparing The Distribution Graph"
                size={30}
              />
            </div>
          </Placeholder>
        }
        options={
          Object {
            "hAxis": Object {
              "title": "Topics",
            },
            "pieHole": 0.5,
            "sliceVisibilityThreshold": 0.005,
            "title": "Topic Distribution",
            "vAxis": Object {
              "title": "Volume (Count)",
            },
          }
        }
        render={null}
        rootProps={Object {}}
        rows={null}
        spreadSheetQueryParameters={
          Object {
            "gid": 1,
            "headers": 1,
          }
        }
        spreadSheetUrl={null}
        style={
          Object {
            "paddingRight": "30px",
          }
        }
        toolbarID={null}
        toolbarItems={null}
        width="50vw"
      />
    </div>
    <div
      style={
        Object {
          "overflowWrap": "anywhere",
          "padding": "85px 10px 10px 10px",
          "position": "relative",
        }
      }
    >
      <Connect(WordCloud) />
    </div>
  </div>
</div>
`;

exports[`<TopicDistributionView /> Snapshots renders Granular distribution for the selectedRollup cluster 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "0px 3px",
      }
    }
  >
    <div
      style={Object {}}
    >
      <Chart
        chartActions={null}
        chartEvents={
          Array [
            Object {
              "callback": [Function],
              "eventName": "ready",
            },
            Object {
              "callback": [Function],
              "eventName": "select",
            },
          ]
        }
        chartType="PieChart"
        chartWrapperParams={Object {}}
        className=""
        columns={null}
        controls={null}
        data={
          Array [
            Array [
              "Topic",
              "Volume",
              Object {
                "role": "wordCloudTerms",
              },
            ],
            Array [
              "cancel_order_service",
              400,
              undefined,
            ],
            Array [
              "cancel_order_customer_service",
              230,
              undefined,
            ],
            Array [
              "store_close_holiday",
              218,
              undefined,
            ],
            Array [
              "size_and_position",
              142,
              undefined,
            ],
          ]
        }
        diffdata={null}
        formatters={null}
        getChartEditor={null}
        getChartWrapper={[Function]}
        graphID={null}
        graph_id={null}
        height="calc(100vh - 40px - 90px - 67px - 10px)"
        legendToggle={false}
        legend_toggle={false}
        loader={
          <Placeholder
            styleOverride={Object {}}
          >
            <div
              className="message-default"
            >
              <LoadingIndicator
                color="#A7A9AF"
                isHerbie={true}
                message="Preparing The Distribution Graph"
                size={30}
              />
            </div>
          </Placeholder>
        }
        options={
          Object {
            "hAxis": Object {
              "title": "Topics",
            },
            "pieHole": 0.5,
            "sliceVisibilityThreshold": 0.005,
            "title": "Topic Distribution",
            "vAxis": Object {
              "title": "Volume (Count)",
            },
          }
        }
        render={null}
        rootProps={Object {}}
        rows={null}
        spreadSheetQueryParameters={
          Object {
            "gid": 1,
            "headers": 1,
          }
        }
        spreadSheetUrl={null}
        style={
          Object {
            "paddingRight": "30px",
          }
        }
        toolbarID={null}
        toolbarItems={null}
        width="50vw"
      />
    </div>
    <div
      style={
        Object {
          "overflowWrap": "anywhere",
          "padding": "85px 10px 10px 10px",
          "position": "relative",
        }
      }
    >
      <Connect(WordCloud) />
    </div>
  </div>
</div>
`;

exports[`<TopicDistributionView /> Snapshots renders Rollup Distribution when there are clusters yet 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "0px 3px",
      }
    }
  >
    <div
      style={Object {}}
    >
      <Chart
        chartActions={null}
        chartEvents={
          Array [
            Object {
              "callback": [Function],
              "eventName": "ready",
            },
            Object {
              "callback": [Function],
              "eventName": "select",
            },
          ]
        }
        chartType="PieChart"
        chartWrapperParams={Object {}}
        className=""
        columns={null}
        controls={null}
        data={
          Array [
            Array [
              "Topic",
              "Volume",
              Object {
                "role": "wordCloudTerms",
              },
            ],
            Array [
              "cancel_order_service",
              400,
              undefined,
            ],
            Array [
              "cancel_order_customer_service",
              230,
              undefined,
            ],
            Array [
              "store_close_holiday",
              218,
              undefined,
            ],
            Array [
              "size_and_position",
              142,
              undefined,
            ],
          ]
        }
        diffdata={null}
        formatters={null}
        getChartEditor={null}
        getChartWrapper={[Function]}
        graphID={null}
        graph_id={null}
        height="calc(100vh - 40px - 90px - 67px - 10px)"
        legendToggle={false}
        legend_toggle={false}
        loader={
          <Placeholder
            styleOverride={Object {}}
          >
            <div
              className="message-default"
            >
              <LoadingIndicator
                color="#A7A9AF"
                isHerbie={true}
                message="Preparing The Distribution Graph"
                size={30}
              />
            </div>
          </Placeholder>
        }
        options={
          Object {
            "hAxis": Object {
              "title": "Topics",
            },
            "pieHole": 0.5,
            "sliceVisibilityThreshold": 0.005,
            "title": "Topic Distribution",
            "vAxis": Object {
              "title": "Volume (Count)",
            },
          }
        }
        render={null}
        rootProps={Object {}}
        rows={null}
        spreadSheetQueryParameters={
          Object {
            "gid": 1,
            "headers": 1,
          }
        }
        spreadSheetUrl={null}
        style={
          Object {
            "paddingRight": "30px",
          }
        }
        toolbarID={null}
        toolbarItems={null}
        width="50vw"
      />
    </div>
    <div
      style={
        Object {
          "overflowWrap": "anywhere",
          "padding": "85px 10px 10px 10px",
          "position": "relative",
        }
      }
    >
      <Connect(WordCloud) />
    </div>
  </div>
</div>
`;

exports[`<TopicDistributionView /> Snapshots renders correctly when finalized filter is applied 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "0px 3px",
      }
    }
  >
    <div
      style={Object {}}
    >
      <Chart
        chartActions={null}
        chartEvents={
          Array [
            Object {
              "callback": [Function],
              "eventName": "ready",
            },
            Object {
              "callback": [Function],
              "eventName": "select",
            },
          ]
        }
        chartType="PieChart"
        chartWrapperParams={Object {}}
        className=""
        columns={null}
        controls={null}
        data={
          Array [
            Array [
              "Topic",
              "Volume",
              Object {
                "role": "wordCloudTerms",
              },
            ],
            Array [
              "cancel_order_service",
              400,
              undefined,
            ],
            Array [
              "cancel_order_customer_service",
              230,
              undefined,
            ],
            Array [
              "store_close_holiday",
              218,
              undefined,
            ],
            Array [
              "size_and_position",
              142,
              undefined,
            ],
          ]
        }
        diffdata={null}
        formatters={null}
        getChartEditor={null}
        getChartWrapper={[Function]}
        graphID={null}
        graph_id={null}
        height="calc(100vh - 40px - 90px - 67px - 10px)"
        legendToggle={false}
        legend_toggle={false}
        loader={
          <Placeholder
            styleOverride={Object {}}
          >
            <div
              className="message-default"
            >
              <LoadingIndicator
                color="#A7A9AF"
                isHerbie={true}
                message="Preparing The Distribution Graph"
                size={30}
              />
            </div>
          </Placeholder>
        }
        options={
          Object {
            "hAxis": Object {
              "title": "Topics",
            },
            "pieHole": 0.5,
            "sliceVisibilityThreshold": 0.005,
            "title": "Topic Distribution",
            "vAxis": Object {
              "title": "Volume (Count)",
            },
          }
        }
        render={null}
        rootProps={Object {}}
        rows={null}
        spreadSheetQueryParameters={
          Object {
            "gid": 1,
            "headers": 1,
          }
        }
        spreadSheetUrl={null}
        style={
          Object {
            "paddingRight": "30px",
          }
        }
        toolbarID={null}
        toolbarItems={null}
        width="50vw"
      />
    </div>
    <div
      style={
        Object {
          "overflowWrap": "anywhere",
          "padding": "85px 10px 10px 10px",
          "position": "relative",
        }
      }
    >
      <Connect(WordCloud) />
    </div>
  </div>
</div>
`;

exports[`<TopicDistributionView /> Snapshots renders correctly when open filter is applied 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "0px 3px",
      }
    }
  >
    <div
      style={Object {}}
    >
      <Chart
        chartActions={null}
        chartEvents={
          Array [
            Object {
              "callback": [Function],
              "eventName": "ready",
            },
            Object {
              "callback": [Function],
              "eventName": "select",
            },
          ]
        }
        chartType="PieChart"
        chartWrapperParams={Object {}}
        className=""
        columns={null}
        controls={null}
        data={
          Array [
            Array [
              "Topic",
              "Volume",
              Object {
                "role": "wordCloudTerms",
              },
            ],
            Array [
              "cancel_order_service",
              400,
              undefined,
            ],
            Array [
              "cancel_order_customer_service",
              230,
              undefined,
            ],
            Array [
              "store_close_holiday",
              218,
              undefined,
            ],
            Array [
              "size_and_position",
              142,
              undefined,
            ],
          ]
        }
        diffdata={null}
        formatters={null}
        getChartEditor={null}
        getChartWrapper={[Function]}
        graphID={null}
        graph_id={null}
        height="calc(100vh - 40px - 90px - 67px - 10px)"
        legendToggle={false}
        legend_toggle={false}
        loader={
          <Placeholder
            styleOverride={Object {}}
          >
            <div
              className="message-default"
            >
              <LoadingIndicator
                color="#A7A9AF"
                isHerbie={true}
                message="Preparing The Distribution Graph"
                size={30}
              />
            </div>
          </Placeholder>
        }
        options={
          Object {
            "hAxis": Object {
              "title": "Topics",
            },
            "pieHole": 0.5,
            "sliceVisibilityThreshold": 0.005,
            "title": "Topic Distribution",
            "vAxis": Object {
              "title": "Volume (Count)",
            },
          }
        }
        render={null}
        rootProps={Object {}}
        rows={null}
        spreadSheetQueryParameters={
          Object {
            "gid": 1,
            "headers": 1,
          }
        }
        spreadSheetUrl={null}
        style={
          Object {
            "paddingRight": "30px",
          }
        }
        toolbarID={null}
        toolbarItems={null}
        width="50vw"
      />
    </div>
    <div
      style={
        Object {
          "overflowWrap": "anywhere",
          "padding": "85px 10px 10px 10px",
          "position": "relative",
        }
      }
    >
      <Connect(WordCloud) />
    </div>
  </div>
</div>
`;

exports[`<TopicDistributionView /> Snapshots renders correctly when there are no clusters yet 1`] = `
<div
  style={
    Object {
      "minWidth": "700px",
    }
  }
>
  <withTheme(ContextualActionsBar)>
    <withTheme(ContextualActionItem)
      icon={[Function]}
      onClickAction={[Function]}
    >
      VIEW BAR CHART
    </withTheme(ContextualActionItem)>
    <withTheme(ContextualActionItem)
      right={true}
    />
    <withTheme(ContextualActionItem)
      disabled={false}
      icon={[Function]}
      onClickAction={[Function]}
      right={true}
    >
      DOWNLOAD
    </withTheme(ContextualActionItem)>
  </withTheme(ContextualActionsBar)>
  <Placeholder
    styleOverride={Object {}}
  >
    <div
      className="message-default"
    >
      <LoadingIndicator
        color="#A7A9AF"
        isHerbie={true}
        message="Preparing The Distribution Graph"
        size={30}
      />
    </div>
  </Placeholder>
</div>
`;
